phoenix-core/src/main/java/org/apache/phoenix/schema/TableRef.java

```diff
@@ -101,7 +105,7 @@ else if (SchemaUtil.isPKColumn(column)) {
     public int hashCode() {
         final int prime = 31;
         int result = alias == null ? 0 : alias.hashCode();
-        result = prime * result + this.table.getName().getString().hashCode();
+        result = prime * result + ( this.table!=null && this.table.getName()!=null ? this.table.getName().getString().hashCode() : 0);
```

Comment:
Good catch, I think a test was failing because TableRef was null, I will add the check to equals.